### PR TITLE
Mark csi-driver-node as node-critical component 

### PR DESCRIPTION
**Pull request by dergeberl**
**How to categorize this PR?**

/area robustness
/kind enhancement
/platform openstack

**What this PR does / why we need it**:

This PR adds the `node.gardener.cloud/critical-component` label to `csi-driver-node` introduced in https://github.com/gardener/gardener/pull/7406

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/7117

**Special notes for your reviewer**:
https://github.com/gardener/gardener/pull/7406 is merged

**Release note**:

```feature user
`csi-driver-node` is marked as a node-critical component. With this, workload pods are only scheduled to a `Node` if it runs a ready `csi-driver-node` pod.
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @dergeberl for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: charts/internal/shoot-system-components/charts/csi-driver-node/templates/daemonset.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ spec:
       annotations:
         checksum/secret-cloud-provider-config: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
       labels:
+        node.gardener.cloud/critical-component: "true"
```
Comment (timebertt) [inline charts/internal/shoot-system-components/charts/csi-driver-node/templates/daemonset.yaml:20]: The label needs to be added to the `DaemonSet` metadata as well now, see https://github.com/gardener/gardener/issues/7117#issuecomment-1425346730.

**Review by timebertt — CHANGES_REQUESTED** [review]

**Comment by timebertt** [comment]
/unhold
now that https://github.com/gardener/gardener/pull/7406 is merged

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Comment by kon-angelo** [comment]
@timebertt PTAL :)

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**